test/unit/UtilsTest.php

```diff
@@ -75,10 +75,14 @@ public function testSerialisation()
             'b' => new core_kernel_classes_Resource('doesnotexist'),
             'c' => array(
                 '1', '2', array(common_user_auth_Service::singleton())
-            )
+            ),
+            'd' => 'aaaaa'.PHP_EOL.'bbbbb'.PHP_EOL.'ccccc'
         );
         $value = eval("return " . common_Utils::toPHPVariableString($toSerialize) . ";");
         $this->assertEquals($toSerialize, $value);
+
+        $valueNice = eval("return " . common_Utils::toHumanReadablePhpString($toSerialize) . ";");
```

Comment:
Ouch, my eyes.... :-D
Why the `eval` here?
What about this ?
```
$valueNice = common_Utils::toHumanReadablePhpString($toSerialize);
```

Reply:
We need to ensure that if the string is evaluated it will result in the same value as the original. How else to test this?

Reply:
You're right, here eval is used to ensure that the change doesn't affect the correctness of the modified code.